jaeger-extension-native/src/main/java/io/ballerina/observe/trace/jaeger/JaegerTracerProvider.java

```diff
@@ -46,34 +51,56 @@ public void init() {    // Do Nothing
-                .withParam(samplerParam.value());
-
-        // Create Sender Configuration
-        Configuration.SenderConfiguration senderConfiguration = new Configuration.SenderConfiguration()
-                .withAgentHost(agentHostname.getValue())
-                .withAgentPort(agentPort);
+                                                BDecimal samplerParam, int reporterFlushInterval,
+                                                int reporterBufferSize) {
+
         String reporterEndpoint = agentHostname + ":" + agentPort;
+        if (!reporterEndpoint.startsWith("http")) {
+            reporterEndpoint = "http://" + reporterEndpoint;
+        }
+
+        JaegerThriftSpanExporter exporter =
+                JaegerThriftSpanExporter.builder()
+                        .setEndpoint(reporterEndpoint)
+                        .build();
+
+        SdkTracerProviderBuilder tracerProviderBuilder = SdkTracerProvider.builder()
+                .addSpanProcessor(BatchSpanProcessor
+                        .builder(exporter)
+                        .setMaxExportBatchSize(reporterBufferSize)
+                        .setExporterTimeout(reporterFlushInterval, TimeUnit.MILLISECONDS)
+                        .build());
 
-        // Create Reporter Configuration
-        reporterConfiguration = new Configuration.ReporterConfiguration()
-                .withSender(senderConfiguration)
-                .withFlushInterval(reporterFlushInterval)
-                .withMaxQueueSize(reporterBufferSize);
+        switch (samplerType.getValue()) {
+            default:
+            case "const":
+                if (samplerParam.value().intValue() == 0) {
+                    tracerProviderBuilder.setSampler(Sampler.alwaysOff());
+                } else {
+                    tracerProviderBuilder.setSampler(Sampler.alwaysOn());
+                }
+                break;
+            case "probabilistic":
+                tracerProviderBuilder.setSampler(Sampler.traceIdRatioBased(samplerParam.value().doubleValue()));
+                break;
+            case "ratelimiting":
+                tracerProviderBuilder.setSampler(new RateLimitingSampler(samplerParam.value().intValue()));
+                break;
+        }
+
+        tracerProvider = tracerProviderBuilder.build();
         console.println("ballerina: started publishing traces to Jaeger on " + reporterEndpoint);
     }
 
     @Override
     public Tracer getTracer(String serviceName) {
-        return new Configuration(serviceName)
-                .withSampler(samplerConfiguration)
-                .withReporter(reporterConfiguration)
-                .getTracerBuilder()
-                .withScopeManager(NoopTracerFactory.create().scopeManager())
-                .build();
+
+        return tracerProvider.get(serviceName);
+    }
+
+    @Override
+    public ContextPropagators getPropagators() {
+
+        return ContextPropagators.create(W3CTraceContextPropagator.getInstance());
```

Comment:
I think Jaeger uses B3 propagator normally.

Reply:
There is a separate [JaegerPropagator](https://github.com/open-telemetry/opentelemetry-java/blob/main/extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/JaegerPropagator.java#L35) implementation as well

Reply:
Fixed